pkg/apis/hive/v1alpha1/clusterdeployment_types.go

```diff
@@ -169,6 +169,11 @@ type ClusterDeploymentStatus struct {
 	// this ClusterDeployment.
 	FederatedClusterRef *corev1.ObjectReference `json:"federatedClusterRef,omitempty"`
 
+	// RawAdminKubeconfigSecret references the secret containing the original admin kubeconfig for this cluster.
+	// This is the kubeconfig as generated by the installer. Use the secret referenced by AdminKubeconfigSecret
```

Comment:
I would add "This kubeconfig will not be functional by itself if you are using additional certificate authorities, and in general it is safer to use the secret referenced by AdminKubeconfigSecret which will include these additional CAs."

Crazy thought here though, instead of exposing this distinction in the API, it's probably a bad idea to just have one secret with the original in an annotation? Probably bypasses some secret storage benefits I assume.

Reply:
Hmm, it's not a bad idea. We could even add another key to the secret instead of just an annotation. Let's discuss at scrum.